rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/LocalDataTransferServer.java

```diff
@@ -213,10 +213,6 @@ public void run()
           throw new Exception("Could not create BookKeeper Client " + Throwables.getStackTraceAsString(e));
         }
 
-        if (!bookKeeperClient.readData(remotePath, offset, readLength, header.getFileSize(), header.getLastModified(), header.getClusterType())) {
-          throw new Exception("Could not cache data required by non-local node");
-        }
-
         if (!CacheConfig.isParallelWarmupEnabled(conf)) {
```

Comment:
Shouldnt you rather remove this if clause? With this change parallel warmup will not add anything to queue

Reply:
Yes.. If parallel warmup is set to true, you call NonLocaReadrequest for the data which is already downloaded. So only invoke readdata (which in turn will put the request in the queue) if parallel warmup is not enabled.
I know this is a little convoltuted way of doing it and we will revisit this piece of code to make it easier to comprehend.

By the way, this change occurred due to some merging problem. Both the lines got merged instead of one

Reply:
Ok, I am forgetting the parallel warmup path. We need to have this info documented somewhere, parallel warmup related classes seems the best place to document this flow.

Reply:
Agree to that. We want to do some refactoring around this area. We can definitely add documentation